CRM/ManualDirectDebit/Form/SignUp.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+use CRM_ManualDirectDebit_ExtensionUtil as E;
+
+/**
+ * Form Manual Direct Debit Sign up Form
+ *
+ */
+class CRM_ManualDirectDebit_Form_SignUp extends CRM_Core_Form {
+
+  private $contributionId;
+
+  public function preProcess() {
+    parent::preProcess();
+    $this->contributionId = CRM_Utils_Request::retrieveValue('contribution_id', 'String', NULL);
```

Comment:
May be better to check if the contribution ID is a positive integer:

```
$this->contributionId = CRM_Utils_Request::retrieveValue('cid', 'Positive', NULL);
```

Reply:
@MiyaNoctem  fixed.